CODE/Sources/PennyBOT/Handlers/InteractionHandler.swift

```diff
@@ -1,104 +1,182 @@
-import Logging
 import DiscordBM
+import Logging
 
 struct InteractionHandler {
-    let discordClient: any DiscordClient
-    let logger: Logger
+    var logger: Logger
     let event: Interaction
+    var pingsService: AutoPingsService {
+        ServiceFactory.makePingsService()
+    }
+    
+    init(logger: Logger, event: Interaction) {
+        self.event = event
+        self.logger = logger
+        self.logger[metadataKey: "event"] = "\(event)"
+    }
     
```

Comment:
The rest of the file id basically just adding the slash commands for auto-pings, as well as refining the `link` command.